### DISCUSSION TITLE
Return the primary key in the search response

### DISCUSSION THREAD

**Discussion by curquiza** [meilisearch/product#186] [Feedback & Feature Proposal]
The program calling MeiliSearch and getting the primary is not always aware of the primary key of the documents.

This is something we particularly need to know the unique key of the documents we are manipulating, especially for the @meilisearch/integration-team when coding integrations.

This is also useful in front-end. For example, React asked for a unique key when displaying a list in front end.

**Comment by bidoubiwa** [comment]
Is this something we want in the search response route or maybe in the `GET /indexes/:index` route. It might bring security issues as it would be made accessible to the `public` API key. I was wondering what you think about it.

**Reply by curquiza** [reply]
If we don't know the primary key in the search response, and if the primary key is not exactly `id`, we have to ask the primary key in parameter to the user, when using the integration, for example in the Rails integration. This adds a step that the integrations could do by default. The primary key is not obvious for everyone that's why I would like to avoid the user understanding it.

(Also, we can do two HTTP calls, one to `/search` and another one to `/indexes/:uid` but no one wants to do that I guess haha.)

From my POV, knowing the primary key in the search response does not involve any security issue, but I'm not an expert.
